File: java/BillsApi/src/main/java/com/apilangarena/javabillsapi/infrastructure/persistence/BillReadRepositoryImpl.java
package com.apilangarena.javabillsapi.infrastructure.persistence;

import com.apilangarena.javabillsapi.application.bills.get.BillReadRepository;
import com.apilangarena.javabillsapi.application.bills.get.BillSummaryDto;
import com.apilangarena.javabillsapi.infrastructure.persistence.repository.BillJpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public class BillReadRepositoryImpl implements BillReadRepository {
    private final BillJpaRepository billJpaRepository;

    public BillReadRepositoryImpl(BillJpaRepository billJpaRepository) {
        this.billJpaRepository = billJpaRepository;
    }

    @Override
    public List<BillSummaryDto> listBills() {
        return billJpaRepository.listBillSummaries().stream()
            .map(item -> new BillSummaryDto(
                item.getId(),
                item.getBillNumber(),
                item.getIssuedAt(),
                item.getTotal(),
                item.getCurrency()
            ))
            .toList();
    }
}

File: kotlin/BillsApi/src/main/kotlin/com/apilangarena/billsapi/infrastructure/persistence/BillReadRepositoryImpl.kt
package com.apilangarena.billsapi.infrastructure.persistence

import com.apilangarena.billsapi.application.bills.get.BillReadRepository
import com.apilangarena.billsapi.application.bills.get.BillSummaryDto
import com.apilangarena.billsapi.infrastructure.persistence.repository.BillJpaRepository
import org.springframework.stereotype.Repository

@Repository
class BillReadRepositoryImpl(
    private val billJpaRepository: BillJpaRepository,
) : BillReadRepository {
    override fun listBills(): List<BillSummaryDto> {
        return billJpaRepository.listBillSummaries().map {
            BillSummaryDto(
                id = it.id,
                billNumber = it.billNumber,
                issuedAt = it.issuedAt,
                total = it.total,
                currency = it.currency,
            )
        }
    }
}
